### PR TITLE
feat: Add custom authentication doc.

### PR DESCRIPTION
**Pull request by mahikhond**
Details:
- Added custom authentication documentation for Stash and Gitlab.

### CODE REVIEW

File: up-and-running/auth/custom.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Custom Authentication
+
+If authentication system use in your organization is not listed in [npme's supported authentication methods](/up-and-running/auth/#authentication) you can use custom authentication, we have provided following two custom authentication plugins:
```
Comment (bcoe) [inline up-and-running/auth/custom.md]: `If authentication system use in your organization` -> `If your authentication system is not listed in`

`you can use custom authentication, we have provided following two custom authentication plugins:` -> `you may be able to an external open-source plugin (or write your own)`. Here are some of the OSS plugins that exist:

File: up-and-running/auth/custom.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * npme-auth-atlassian-stash
```
Comment (bcoe) [inline up-and-running/auth/custom.md]: let's make these links to the repos.

File: up-and-running/auth/custom.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * npme-auth-gitlab
+
+You can also opt to write your own `custom auth-plugin` for npm Enterprise.
```
Comment (bcoe) [inline up-and-running/auth/custom.md]: maybe add, `using one of the open-source plugins listed above as a starting point`.

**Review by bcoe — CHANGES_REQUESTED** [review]
this is sounding great 👍 I'm glad to have these plugins documented.

**Comment by mahikhond** [comment]
@bcoe Updated requested changes

File: up-and-running/auth/custom.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# Custom Authentication
+
+If your authentication system is not listed in [npme's supported authentication methods](/up-and-running/auth/#authentication) you may be able to use an external open-source plugin (or write your own). Here are some of the SSO plugins that exist:
```
Comment (bcoe) [inline up-and-running/auth/custom.md]: `Here are some of the SSO plugins that exist` -> `Here are some of the existing third-party plugins that exist`

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
👍 thanks for the contribution.

**Pull request merged**